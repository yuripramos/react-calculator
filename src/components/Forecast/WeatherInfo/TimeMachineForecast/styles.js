import styled, { css } from "styled-components";
import { rem, media } from "../../../../styles/tools";

export const InnerWrapper = styled.div`
  text-align: center;
  min-height: ${rem(250)};
  display: inline-flex;
  flex-wrap: wrap;
`;
