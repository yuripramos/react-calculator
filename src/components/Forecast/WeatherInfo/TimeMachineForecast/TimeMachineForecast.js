import React, { Component } from "react";
import moment from "moment";
import { arrayOf, shape, number, string, bool, func } from "prop-types";

import IconGenerator from "../../../common/Icon";
import { translate } from "../../../../utils/i18n";
import {
  ContentWrapper,
  Temperature,
  TinySpecs,
  Icon,
  DaySpecs
} from "../FutureForecast/styles";
import { InnerWrapper } from "./styles";
import { Container, Row, Column } from "../../../../styles/grid";

class FutureForecast extends Component {
  componentDidUpdate(prevProps) {
    const { isTimeMachineActive, clearSearch } = this.props;
    if (isTimeMachineActive !== prevProps.isTimeMachineActive) {
      clearSearch();
    }
  }
  render() {
    const { forecast } = this.props;
    return (
      <ContentWrapper>
        <Container>
          <Row>
            <Column>
              <InnerWrapper>
                {forecast.map((e, i) => (
                  <DaySpecs key={i} tiny>
                    <TinySpecs tiny>
                      {moment.unix(e.currently.time).format("L")}
                    </TinySpecs>
                    <Icon>
                      <IconGenerator
                        name={e.currently.icon}
                        width={"50px"}
                        height={"50px"}
                      />
                    </Icon>
                    <Temperature tiny>
                      {`${e.currently.temperature} ${translate("TEMPERATURE")}`}{" "}
                    </Temperature>
                  </DaySpecs>
                ))}
              </InnerWrapper>
            </Column>
          </Row>
        </Container>
      </ContentWrapper>
    );
  }
}

FutureForecast.defaultProps = {};

FutureForecast.propTypes = {
  forecast: arrayOf(
    shape({
      time: number,
      icon: string,
      temperature: number
    })
  ),
  isTimeMachineActive: bool,
  clearSearch: func,
};

export default FutureForecast;
