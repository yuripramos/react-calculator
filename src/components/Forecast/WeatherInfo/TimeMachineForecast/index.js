import TimeMachineForecast from "./TimeMachineForecast";

export default TimeMachineForecast;
