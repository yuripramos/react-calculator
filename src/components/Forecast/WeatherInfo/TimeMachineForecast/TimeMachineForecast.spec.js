import React from "react";
import { shallow } from "enzyme";


import TimeMachineForecast from "./TimeMachineForecast";

let props;

beforeEach(() => {
  props = {
    forecast: [
      {
        currently: {
          time: 214124,
          icon: "fog",
          temperature: 20
        }
      },
      {
        currently: {
          time: 123124,
          icon: "snow",
          temperature: -10
        }
      }
    ],
    isTimeMachineActive: true,
    clearSearch: jest.fn(),
    getForecastTimeMachine: jest.fn(),
    toggleForecast: jest.fn()
  };
});

describe("TimeMachineForecast Component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<TimeMachineForecast {...props} />)).toMatchSnapshot();
  });

});
