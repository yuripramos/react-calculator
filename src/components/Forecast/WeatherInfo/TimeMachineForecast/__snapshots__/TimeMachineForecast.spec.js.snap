// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeMachineForecast Component should match snapshot with props 1`] = `
<styled.div>
  <Container>
    <Row>
      <Column>
        <styled.div>
          <styled.div
            key="0"
            tiny={true}
          >
            <styled.span
              tiny={true}
            >
              01/01/2010
            </styled.span>
            <styled.div>
              <IconGenerator
                height="50px"
                name="fog"
                width="50px"
              />
            </styled.div>
            <styled.span
              tiny={true}
            >
              20 undefined
               
            </styled.span>
          </styled.div>
          <styled.div
            key="1"
            tiny={true}
          >
            <styled.span
              tiny={true}
            >
              01/01/2010
            </styled.span>
            <styled.div>
              <IconGenerator
                height="50px"
                name="snow"
                width="50px"
              />
            </styled.div>
            <styled.span
              tiny={true}
            >
              -10 undefined
               
            </styled.span>
          </styled.div>
        </styled.div>
      </Column>
    </Row>
  </Container>
</styled.div>
`;
